### ISSUE TITLE
Server shutdown is not detected when mod fails to download

### ISSUE THREAD

**Issue by taikodragon** [SEServerExtender/SEServerExtender#41]
When starting the SE server if one of the mods fails to download (such as being removed from the workshop), SESE will not detect that the SE server has decided to shutdown and will attempt to read data from the process, causing the faction refresh to fail with a NullReferenceException.

**Comment by dodexahedron** [comment]
I think I saw that behavior a long time ago, too.
I'll take a look.

**Comment by hkraal** [comment]
I saw it last Thursday due steam communities being down for a few minutes.

**Comment by markam67** [comment]
Yes this has happened repeatedly to me over the last few weeks. Whenever a mod fails the server stops the action to load the world but still reports errors and dose not actually stop. I really wish we could change it so that if a mod fails to download the server extender would do two things 
1. Drop a message in the log on the last line of the log ( or as close as possible) that that is why it failed. 
2. longer term it would be nice to add the functionality to the server extender to keep a list of the mods with their information attached ( name, description, number) and only replace the list on a successful load so that when a mod fails to load it can compare to the list and report the failing mod. this servers 2 purposes, to let the server admin know what happened and to remove the offending mod number, and to let the server users know what blocks or features just went missing.
